### PR TITLE
plot trend

### LINKED ISSUE #515: Trend Plot
### 🚀 Feature Request

Create the plot to visualize trend

### Motivation

Tool for trend analysis

### Proposal

In etna.analysis.plotters create method: 
```
def plot_trend(
    ts: "TSDataset",
    trend_transform: "TrendTransformType",
    segments: Optional[List[str]] = None,
    columns_num: int = 2,
    figsize: Tuple[int, int] = (10, 5),
):
```
Where
trend_transform - instance of one of the transforms from etna.transforms.decomposition

1. Apply trend_transform to the copy of ts
2. Obtain the trend = ts[in_column] - transformed_ts[in_column], where in_column is obtained from trend_transform
3. Create scatter plot of the series values with the trend line
4. If trend_transform is instance of Union[LinearTrendTransform, TheilSenTrendTransform] add equation of the trend line to the legend(use coefficients of the linear model) 

`TrendTransformType = Union[ChangePointsTrendTransform, LinearTrendTransform, TheilSenTrendTransform, STLTransform ]`
Add picture with visualization example to PR
### Test cases

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [ ] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by Ama16**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Closing issues

closes #515

### CODE REVIEW

**Comment by Ama16** [comment]
Снимок экрана 2022-02-22 в 12 23 49
Снимок экрана 2022-02-22 в 12 24 09
Снимок экрана 2022-02-22 в 12 24 24
Снимок экрана 2022-02-22 в 12 24 38

File: etna/analysis/plotters.py [hunk 1]
```diff
@@ -23,6 +23,10 @@
 
 if TYPE_CHECKING:
     from etna.datasets import TSDataset
+    from etna.transforms import ChangePointsTrendTransform
+    from etna.transforms import LinearTrendTransform
+    from etna.transforms import STLTransform
+    from etna.transforms import TheilSenTrendTransform
```
Comment (iKintosh) [inline etna/analysis/plotters.py:29]: Why are these lines here?

File: etna/analysis/plotters.py [hunk 2]
```diff
@@ -729,3 +733,59 @@ def plot_residuals(
...
+        labels_short = [i[: i.find("(")] for i in labels]
+        if len(np.unique(labels_short)) == len(labels_short):
+            labels = labels_short
+    else:
+        df_detrend = [trend_transform.fit_transform(df.copy())]
+        labels = [trend_transform.__repr__()[: trend_transform.__repr__().find("(")]]
+        if isinstance(trend_transform, LinearTrendTransform) or isinstance(trend_transform, TheilSenTrendTransform):
+            for seg in segments:
+
+                linear_coeffs[seg] = ", k=" + str(trend_transform.segment_transforms[seg]._linear_model.coef_[0])
```
Comment (iKintosh) [inline etna/analysis/plotters.py:783]: Does it mean, that we can see coefficient of linear model only if we use it alone?

File: etna/analysis/plotters.py [hunk 3]
```diff
@@ -729,3 +733,59 @@ def plot_residuals(
...
+
+    ax = prepare_axes(segments=segments, columns_num=columns_num, figsize=figsize)
+    df = ts.df
+    linear_coeffs = dict(zip(segments, ["" for i in range(len(segments))]))
+    if isinstance(trend_transform, list):
+        df_detrend = [transform.fit_transform(df.copy()) for transform in trend_transform]
+        labels = [transform.__repr__() for transform in trend_transform]
+        labels_short = [i[: i.find("(")] for i in labels]
+        if len(np.unique(labels_short)) == len(labels_short):
+            labels = labels_short
```
Comment (iKintosh) [inline etna/analysis/plotters.py:776]: This seems like we need this "if" only in some cases, because some transforms do not have short labels. Right?

**Review by iKintosh — CHANGES_REQUESTED** [review]

File: etna/analysis/plotters.py (hunk 3 above)
Comment (iKintosh) [inline etna/analysis/plotters.py:776]: Let's make it a private method, that constructs labels. 
It should have small docsting on how it works:
```
If only unique transform classes then show their short names (without parameters)
If  not, then show their full repr as label
```

File: etna/analysis/plotters.py [hunk 4]
```diff
@@ -729,3 +733,59 @@ def plot_residuals(
...
+    else:
+        df_detrend = [trend_transform.fit_transform(df.copy())]
+        labels = [trend_transform.__repr__()[: trend_transform.__repr__().find("(")]]
```
Comment (iKintosh) [inline etna/analysis/plotters.py:779]: Remove this `else` because it repeats the code above.

File: etna/analysis/plotters.py (hunk 2 above)
Comment (iKintosh) [inline etna/analysis/plotters.py:783]: This check should be done afterwards as a separate check.

**Review by iKintosh — CHANGES_REQUESTED** [review]
Added comments on how it should be changed

File: etna/analysis/plotters.py [hunk 5]
```diff
@@ -729,3 +733,69 @@ def plot_residuals(
...
         ax[i].tick_params("x", rotation=45)
         ax[i].set_xlabel(feature)
+
+
+TrendTransformType = Union[
+    "ChangePointsTrendTransform", "LinearTrendTransform", "TheilSenTrendTransform", "STLTransform"
+]
+
+
+def __get_labels_names(trend_transform, segments):
```
Comment (iKintosh) [inline etna/analysis/plotters.py:743]:
```suggestion
def _get_labels_names(trend_transform, segments):
```

[No double ](https://stackoverflow.com/a/1301369/7415703)

File: etna/analysis/plotters.py [hunk 6]
```diff
@@ -729,3 +733,69 @@ def plot_residuals(
...
+        segments = list(set(ts.columns.get_level_values("segment")))
+
+    ax = prepare_axes(segments=segments, columns_num=columns_num, figsize=figsize)
+    df = ts.df
+
+    if not isinstance(trend_transform, list):
+        trend_transform = [trend_transform]
+
+    df_detrend = [transform.fit_transform(df.copy()) for transform in trend_transform]
+    labels, linear_coeffs = __get_labels_names(trend_transform, segments)
```
Comment (iKintosh) [inline etna/analysis/plotters.py:793]:
```suggestion
    labels, linear_coeffs = _get_labels_names(trend_transform, segments)
```
Same as above https://stackoverflow.com/a/1301369/7415703

**Review by iKintosh — CHANGES_REQUESTED** [review]
Just one fix needed

**Comment by Ama16** [comment]
Снимок экрана 2022-03-09 в 11 03 23

**Review by iKintosh — APPROVED** [review]
👍
